spec.html

```diff
@@ -42515,10 +42821,11 @@ <h1>ArrayBuffer.prototype.slice ( _start_, _end_ )</h1>
           1. If IsDetachedBuffer(_new_) is *true*, throw a *TypeError* exception.
           1. If SameValue(_new_, _O_) is *true*, throw a *TypeError* exception.
           1. If _new_.[[ArrayBufferByteLength]] &lt; _newLen_, throw a *TypeError* exception.
-          1. NOTE: Side-effects of the above steps may have detached _O_.
+          1. NOTE: Side-effects of the above steps may have detached or resized _O_.
           1. If IsDetachedBuffer(_O_) is *true*, throw a *TypeError* exception.
           1. Let _fromBuf_ be _O_.[[ArrayBufferData]].
           1. Let _toBuf_ be _new_.[[ArrayBufferData]].
+          1. If _first_ &lt; _O_.[[ArrayBufferByteLength]], set _newLen_ to min(_O_.[[ArrayBufferByteLength]], _newLen_).
```

Comment:
[CopyDataBlockBytes](https://tc39.es/ecma262/#sec-copydatablockbytes) asserts all inputs are in-bounds, so this check needs to be:

```suggestion
            1. Let _currentLen_ be _O_.[[ArrayBufferByteLength]].
            1. If _first_ &lt; _currentLen_, then
              1. Let _count_ be min(_newLen_, _currentLen_ - _first_).
              1. Perform CopyDataBlockBytes(_toBuf_, 0, _fromBuf_, _first_, _count_).
```

I'm also wondering if there should actually be a TypeError thrown when `O` is resized (either if `first < O.[[ArrayBufferByteLength]]` or even `first + count < O.[[ArrayBufferByteLength]]`), because we already throw a TypeError if the newly created ArrayBuffer is too small.

Reply:
Fixed.

> I'm also wondering if there should actually be a TypeError [...]

The design decision for these methods on how to handle the resized-in-the-middle edge cases was to be analogous with how Array methods dealt with resized-in-the-middle cases, which is "compute and cache the loop bounds ahead of time, don't crash, but keep on trucking otherwise". IOW, the lengths need to be reloaded to ensure things don't crash (like this bug you found), but don't bother with throwing.